javaagent-extensions/gcp-auth/src/main/java/com/google/cloud/opentelemetry/extension/auth/GcpAuthAutoConfigurationCustomizerProvider.java

```diff
@@ -0,0 +1,135 @@
+
+  /**
+   * Customizes the provided {@link AutoConfigurationCustomizer}.
+   *
+   * <p>This method attempts to retrieve Google Application Default Credentials (ADC) and performs
+   * the following: - Adds authorization headers to the configured {@link SpanExporter} based on the
+   * retrieved credentials. - Adds default properties for OTLP endpoint and resource attributes for
+   * GCP integration.
+   *
+   * @param autoConfiguration the AutoConfigurationCustomizer to customize.
+   * @throws GoogleAuthException if there's an error retrieving Google Application Default
+   *     Credentials.
+   * @throws io.opentelemetry.sdk.autoconfigure.spi.ConfigurationException if required options are
+   *     not configured through environment variables or system properties.
+   */
+  @Override
+  public void customize(AutoConfigurationCustomizer autoConfiguration) {
+    try {
+      GoogleCredentials credentials = GoogleCredentials.getApplicationDefault();
+      autoConfiguration
+          .addSpanExporterCustomizer(
+              (exporter, configProperties) -> addAuthorizationHeaders(exporter, credentials))
+          .addPropertiesSupplier(this::getRequiredProperties)
+          .addResourceCustomizer(this::customizeResource);
+    } catch (IOException e) {
+      throw new GoogleAuthException(Reason.FAILED_ADC_RETRIEVAL, e);
+    }
+  }
+
+  @Override
+  public int order() {
+    return Integer.MAX_VALUE - 1;
+  }
+
+  // Adds authorization headers to the calls made by the OtlpGrpcSpanExporter and
+  // OtlpHttpSpanExporter.
+  private SpanExporter addAuthorizationHeaders(
+      SpanExporter exporter, GoogleCredentials credentials) {
+    if (exporter instanceof OtlpHttpSpanExporter) {
+      OtlpHttpSpanExporterBuilder builder =
+          ((OtlpHttpSpanExporter) exporter)
+              .toBuilder().setHeaders(() -> getRequiredHeaderMap(credentials));
+      return builder.build();
+    } else if (exporter instanceof OtlpGrpcSpanExporter) {
+      OtlpGrpcSpanExporterBuilder builder =
+          ((OtlpGrpcSpanExporter) exporter)
+              .toBuilder().setHeaders(() -> getRequiredHeaderMap(credentials));
+      return builder.build();
+    }
+    return exporter;
+  }
+
+  private Map<String, String> getRequiredHeaderMap(GoogleCredentials credentials) {
+    Map<String, String> gcpHeaders = new HashMap<>();
+    try {
+      credentials.refreshIfExpired();
+    } catch (IOException e) {
+      throw new GoogleAuthException(Reason.FAILED_ADC_REFRESH, e);
+    }
+    gcpHeaders.put(QUOTA_USER_PROJECT_HEADER, credentials.getQuotaProjectId());
+    gcpHeaders.put("Authorization", "Bearer " + credentials.getAccessToken().getTokenValue());
+    return gcpHeaders;
+  }
+
+  // Sets the required properties that are essential for exporting OTLP data to GCP.
+  private Map<String, String> getRequiredProperties() {
+    Map<String, String> properties = new HashMap<>();
+    properties.put("otel.resource.providers.gcp.enabled", "true");
```

Comment:
Does this mean that exporting to GCP automatically enables the gcp resource detector?  Or just that it sets gcp.project_id.  We probably don't want to enable the gcp resource detector.

Reply:
The GCP resource providers [are now built-in the java agent](https://opentelemetry.io/docs/zero-code/java/agent/configuration/#enable-resource-providers-that-are-disabled-by-default) and controlled through this property.

Including this property here means when users include this extension with their Java app, the property is automatically set to true, so they won't have to do it manually.

Reply:
I wonder if that actually makes sense.  In theory, users might be using google cloud observability without running on GCP.  If the user explicitly sets `otel.resource.providers.gcp.enabled` to false, will that override this?  Seems like ideally we would default to true, but allow users to disable it.

Reply:
I see your point,  based on the [method documentation](https://github.com/open-telemetry/opentelemetry-java/blob/a7a2c061459a8a57fe560ebb0410948f545f8293/sdk-extensions/autoconfigure-spi/src/main/java/io/opentelemetry/sdk/autoconfigure/spi/AutoConfigurationCustomizer.java#L91), it would seem like specifying this here would prevent users from overriding it, so I will just remove this.